pyreisejl/utility/extract_data.py

```diff
@@ -65,9 +65,18 @@ def extract_data(results):
     cost = []
 
     extraction_vars = {"pf", "pg", "lmp", "congu", "congl"}
-    sparse_extraction_vars = {"congu", "congl", "load_shed"}
+    sparse_extraction_vars = {"congu", "congl", "load_shed", "trans_viol"}
     temps = {}
     outputs = {}
+    optional_variables = [
+        {"name": "pf_dcline", "key1": "dcline", "key2": "PF_dcline"},
+        {"name": "storage_pg", "key1": "storage", "key2": "PG"},
+        {"name": "storage_e", "key1": "storage", "key2": "Energy"},
+        {"name": "load_shed", "key1": "load_shed", "key2": "load_shed"},
+        {"name": "load_shift_up", "key1": "flexible_demand", "key2": "load_shift_up"},
+        {"name": "load_shift_dn", "key1": "flexible_demand", "key2": "load_shift_dn"},
+        {"name": "trans_viol", "key1": "trans_viol", "key2": "trans_viol"},
```

Comment:
Is it possible that we replace `key1` and `key2` with more meaningful names? NVM if this is just due to meaningless mat file structure.

Reply:
These are only used within this function to build the `output_mpc` matfile with the structure we want:
```python
for var in optional_variables:
    try:
        temps[var["name"]] = output_mpc[var["key1"]][var["key2"]].T
        extraction_vars.add(var["name"])
    except KeyError:
        pass
```